modules/ballerina-test/src/test/java/org/ballerinalang/test/services/nativeimpl/request/RequestNativeFunctionSuccessTest.java

```diff
@@ -710,17 +771,75 @@ public void testSetBinaryPayload() {
+    @Test (description = "Test setEntityBody() function")
+    public void testSetEntityBody() {
+        BStruct request = BCompileUtil.createAndGetStruct(result.getProgFile(), protocolPackageHttp, requestStruct);
+        HTTPCarbonMessage requestMsg = HttpUtil.createHttpCarbonMessage(true);
+        HttpUtil.addCarbonMsg(request, requestMsg);
 
-        Assert.assertEquals(blobDataSource.getMessageAsString(), "Ballerina", "Payload is not set properly");
+        try {
+            File file = File.createTempFile("test", ".json");
+            file.deleteOnExit();
+            BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file));
+            bufferedWriter.write("{'name':'wso2'}");
+            bufferedWriter.close();
+
+            BStruct fileStruct = BCompileUtil.createAndGetStruct(result.getProgFile(), protocolPackageFile, FILE);
+            fileStruct.setStringField(0, file.getAbsolutePath());
+            BValue[] inputArg = { request, fileStruct, new BString(APPLICATION_JSON) };
+            BValue[] returnVals = BRunUtil.invoke(result, "testSetEntityBody", inputArg);
+            Assert.assertFalse(returnVals == null || returnVals.length == 0 || returnVals[0] == null,
+                    "Invalid Return Values.");
+            Assert.assertTrue(returnVals[0] instanceof BStruct);
+            BStruct entity = (BStruct) ((BStruct) returnVals[0]).getNativeData(MESSAGE_ENTITY);
+            BStruct returnFileStruct = (BStruct) entity.getRefField(OVERFLOW_DATA_INDEX);
+
+            String returnJsonValue = new String (Files.readAllBytes(Paths.get(returnFileStruct.getStringField(0))),
+                    UTF_8);
+            BJSON bJson = new BJSON(returnJsonValue);
+            Assert.assertEquals(bJson.value().get("name").asText(), "wso2", "Payload is not set properly");
+        } catch (IOException e) {
+            LOG.error("Error occured while creating a temporary file in testSetEntityBody", e.getMessage());
+        }
+    }
+
+    /**
+     * Get the response value from input stream.
+     *
+     * @param response carbon response
+     * @return return value from  input stream as a string
+     */
+    private String getReturnValue(HTTPCarbonMessage response) {
+        Reader reader;
+        final int bufferSize = 1024;
+        final char[] buffer = new char[bufferSize];
+        final StringBuilder out = new StringBuilder();
+        try {
+            reader = new InputStreamReader(new HttpMessageDataStreamer(response).getInputStream(), UTF_8);
+            while (true) {
+                int size = reader.read(buffer, 0, buffer.length);
+                if (size < 0) {
+                    break;
+                }
+                out.append(buffer, 0, size);
+            }
+        } catch (UnsupportedEncodingException e) {
+            LOG.error("Error occured while reading the response value in getReturnValue", e.getMessage());
+        } catch (IOException e) {
```

Comment:
Two catch branches are identical. So we don't need explicitly catch UnsupportedEncodingException here